using a const named `POOL_SIZE`  with embassy_executor::task fails to compile

With embassy-executor 0.10.0, this code:
~~~~rust
const POOL_SIZE: usize = 1;

#[embassy_executor::task(pool_size = POOL_SIZE)]
async fn some_task() -> ! {
    unimplemented!();
}
~~~~
fails to compile, with rustc complaining:
~~~~
error[E0391]: cycle detected when checking if `some_task::POOL_SIZE` is a trivial const
   --> src/main.rs:137:1
    |
137 | #[embassy_executor::task(pool_size = POOL_SIZE)]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
note: ...which requires building MIR for `some_task::POOL_SIZE`...
   --> src/main.rs:137:1
    |
137 | #[embassy_executor::task(pool_size = POOL_SIZE)]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: ...which again requires checking if `some_task::POOL_SIZE` is a trivial const, completing the cycle
note: cycle used when building MIR for `some_task::__task_pool_get::{constant#0}`
   --> src/main.rs:137:1
    |
137 | #[embassy_executor::task(pool_size = POOL_SIZE)]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: see https://rustc-dev-guide.rust-lang.org/overview.html#queries and https://rustc-dev-guide.rust-lang.org/query.html for more information
    = note: this error originates in the attribute macro `embassy_executor::task` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0391`.
~~~~

Using any other name than `POOL_SIZE` works fine, so this is not a critical bug or anything, but it'd be nice if that name were available.

--- a/embassy-executor-macros/src/macros/task.rs
+++ b/embassy-executor-macros/src/macros/task.rs
@@ -207,13 +207,13 @@ pub fn run(args: TokenStream, item: TokenStream) -> TokenStream {
             }
         }
 
-        const POOL_SIZE: usize = #pool_size;
-        static POOL: #embassy_executor::raw::TaskPool<<() as _EmbassyInternalTaskTrait>::Fut, POOL_SIZE> = #embassy_executor::raw::TaskPool::new();
+        const __POOL_SIZE: usize = #pool_size;
+        static POOL: #embassy_executor::raw::TaskPool<<() as _EmbassyInternalTaskTrait>::Fut, __POOL_SIZE> = #embassy_executor::raw::TaskPool::new();
         unsafe { POOL.#spawn(move || <() as _EmbassyInternalTaskTrait>::construct(#(#full_args,)*)) }
     };
     #[cfg(not(feature = "nightly"))]
     let mut task_outer_body = quote! {
-        const fn __task_pool_get<F, Args, Fut>(_: F) -> &'static #embassy_executor::raw::TaskPool<Fut, POOL_SIZE>
+        const fn __task_pool_get<F, Args, Fut>(_: F) -> &'static #embassy_executor::raw::TaskPool<Fut, __POOL_SIZE>
         where
             F: #embassy_executor::_export::TaskFn<Args, Fut = Fut>,
             Fut: ::core::future::Future + 'static,
@@ -221,11 +221,11 @@ pub fn run(args: TokenStream, item: TokenStream) -> TokenStream {
             unsafe { &*POOL.get().cast() }
         }
 
-        const POOL_SIZE: usize = #pool_size;
+        const __POOL_SIZE: usize = #pool_size;
         static POOL: #embassy_executor::_export::TaskPoolHolder<
-            {#embassy_executor::_export::task_pool_size::<_, _, _, POOL_SIZE>(#task_inner_ident)},
-            {#embassy_executor::_export::task_pool_align::<_, _, _, POOL_SIZE>(#task_inner_ident)},
-        > = unsafe { ::core::mem::transmute(#embassy_executor::_export::task_pool_new::<_, _, _, POOL_SIZE>(#task_inner_ident)) };
+            {#embassy_executor::_export::task_pool_size::<_, _, _, __POOL_SIZE>(#task_inner_ident)},
+            {#embassy_executor::_export::task_pool_align::<_, _, _, __POOL_SIZE>(#task_inner_ident)},
+        > = unsafe { ::core::mem::transmute(#embassy_executor::_export::task_pool_new::<_, _, _, __POOL_SIZE>(#task_inner_ident)) };
         unsafe { __task_pool_get(#task_inner_ident).#spawn(move || #task_inner_ident(#(#full_args,)*)) }
     };
 

